test/tbb/test_arena_constraints.cpp

```diff
@@ -208,3 +212,12 @@ TEST_CASE("Testing constraints_threads_per_core() reserved entry point") {
     tbb::task_arena::constraints c{};
     tbb::detail::r1::constraints_threads_per_core(c);
 }
+
+//! Using custom assertion handler to test failure on invalid constraints
+//! \brief \ref interface \ref error_guessing
+TEST_CASE("Using custom assertion handler to test failure on invalid constraints") {
+    tbb::set_assertion_handler(utils::AssertionFailureHandler);
+    TRY_BAD_EXPR(tbb::info::default_concurrency(tbb::task_arena::constraints{}.set_max_threads_per_core(0)),
+        "Wrong max_threads_per_core constraints field value.");
+    tbb::set_assertion_handler(nullptr); // Reset to default handler
```

Comment:
I would add the following checks:
* Check that `set_assertion_handler` returns previously installed handler
* Check that `get_assertion_handler` returns correct handler before and after both sets
* Check that `assertion_handler_type` is exactly the type defined in the documentation